ql/src/java/org/apache/hadoop/hive/ql/parse/type/RexNodeExprFactory.java

```diff
@@ -1000,6 +1003,21 @@ protected FunctionInfo getFunctionInfo(String funcName) throws SemanticException
     return functionHelper.getFunctionInfo(funcName);
   }
 
+  @Override
+  protected RexNode replaceFieldNamesInStruct(RexNode expr, List<String> newFieldNames) {
+    if (newFieldNames.isEmpty()) {
+      return expr;
+    }
+
+    RexCall structCall = (RexCall) expr;
+    List<RexNode> newOperands = structCall.operands.stream()
+            .filter(rexNode -> "_UTF-16LE'tok_alias':VARCHAR(2147483647) CHARACTER SET \"UTF-16LE\"".compareTo(rexNode.toString()) != 0)
```

Comment:
Have you considered skipping the token in `TypeCheckProcFactory` rather than here? I think you could do that if it would not extend `StrExprProcessor`. If you cannot, maybe you can return a 'placeholder' object (static final) that you can identify easily with == ?
I think that would simplify the implementation.
Also note that tokens are not handled directly by the `ExprFactory` implementation.

Reply:
It cannot be skipped because `TypeCheckProcFactory` uses `SemanticGraphWalker` which requires a return value other than `null` from `SemanticNodeProcessor.process`.
I introduced a placeholder and filter it out when collecting the children of a function node.